packages/core/analytics/core-analytics-browser-internal/src/track_performance_measure_entries.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import type { AnalyticsClient } from '@kbn/analytics-client';
+import { reportPerformanceMetricEvent } from '@kbn/ebt-tools';
+
+export function trackPerformanceMeasureEntries(analytics: AnalyticsClient, isDevMode: boolean) {
+  function perfObserver(
+    list: PerformanceObserverEntryList,
+    observer: PerformanceObserver,
+    droppedEntriesCount: number
+  ) {
+    list.getEntries().forEach((entry) => {
+      if (entry.entryType === 'measure') {
+        const measureName = entry?.name;
+        const duration = entry.duration;
+
+        if (isDevMode) {
+          if (!measureName) {
+            console.error(`Failed to report the performance entry. Measure name is undefined`);
+          }
+
+          if (!duration) {
+            console.error(
+              `Failed to report the performance entry. Duration for the measure: ${measureName} is undefined`
+            );
+          }
+
+          console.log(`The measure ${measureName} completed in ${duration}ms`);
+        }
+
+        if (droppedEntriesCount > 0) {
+          console.warn(`${droppedEntriesCount} entries got dropped due to the buffer being full.`);
+        }
+
+        try {
+          reportPerformanceMetricEvent(analytics, {
+            eventName: measureName,
+            duration,
+          });
```

Comment:
I wonder if we want to revisit the shape of the event:

Does it make sense to name all events `eventName: 'render' | 'time_to_render' | ...` and then use the `meta` object to identify the component? The content in `meta` could be just one ID like you did here `measureName`, or a full object with `domain/app: 'apm' | 'infra' | ..., view: 'services' | 'hosts' | ..., component: ..., details: ...`

This way we can run analyses like:
* avg or rendering time in Kibana as a whole (just filter by `eventName`)
* avg or rendering time in the app `apm` (`eventName` + `meta.domain/app`)
* top components (the ones taking the longest) => filter by `eventName`, aggregate by `meta.component`

This could also open the possibility of components providing extra info (somehow... to be added in another PR) like the size of data they processed or how long the fetch request took to provide more context for us to understand if it's slow because of the network or because of rendering the data.

WDYT?

Reply:
+1, I think we need take time and discuss the performance event structure.
The original one is described in https://github.com/elastic/kibana/blob/363f4b7583ae68bf6e2818adb51769d170fc5479/dev_docs/tutorials/performance/adding_custom_performance_metrics.mdx#instrument-your-code-to-report-metric-event 

is quite generic and we already have events defined in non-easy to understand way. I also wish to have more aggregation capabilities, e.g. be able to group events related to the same Kibana page.

Reply:
> 
> This way we can run analyses like:
> 
> * avg or rendering time in Kibana as a whole (just filter by `eventName`)
> * avg or rendering time in the app `apm` (`eventName` + `meta.domain/app`)
> * top components (the ones taking the longest) => filter by `eventName`, aggregate by `meta.component`
> 
> This could also open the possibility of components providing extra info (somehow... to be added in another PR) like the size of data they processed or how long the fetch request took to provide more context for us to understand if it's slow because of the network or because of rendering the data.
> 
> WDYT?

That's a really good point @afharo. Looking at the [schema](https://github.com/elastic/kibana/blob/0682fd3ef4b1bd62d1dbb660b91a6d0b282042c6/packages/kbn-ebt-tools/src/performance_metric_events/schema.ts#L20-L22), the meta field isn't aggregatable. However, we could consider creating a runtime field, though it adds a bit of complexity for the analysis in telemetry cluster. and I'm not sure if it affects the performance. 

```
{
   "event_type": "performance_metric",
   "properties": {
      “eventName”: “time_to_render”,
      “meta”: {
          “target”: 'apm.services.list...',  
      }
      "duration" : 3500, // Total duration 
      
      /// FUTURU:  up to 9 key/value pairs
      "key1"  : "time_to_data", 
      "value1": 3000,
      "key2"  : "render_count", 
      "value2": 5,
   }
}

```